actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,11 +92,20 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
+      # This is in the case that the browser client wants
+      # to open the connection regardless of whether a
+      # protocol was negotiated successfully.
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        @close()
+      else
+        @disconnected = false
+        @subscriptions.reload()
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        ActionCable.log("WebSocket is closing because there is no common protocol. Stopping ConnectionMonitor.")
```

Comment:
For debugging, it would be useful to log the client and server protocol versions rather than say they're not common.